annif/backend/backend.py

```diff
@@ -12,13 +12,30 @@ class AnnifBackend(metaclass=abc.ABCMeta):
     needs_subject_index = False
     needs_subject_vectorizer = False
 
+    DEFAULT_PARAMS = {'limit': 100}
+
     def __init__(self, backend_id, params, datadir):
         """Initialize backend with specific parameters. The
         parameters are a dict. Keys and values depend on the specific
         backend type."""
         self.backend_id = backend_id
-        self.params = params
         self.datadir = datadir
+        self.params = self.fill_params_with_defaults(params)
+
+    def default_params(self):
+        return self.DEFAULT_PARAMS
+
+    def fill_params_with_defaults(self, params):
```

Comment:
Would the params actually be better to be implemented as a property, and then this would be decorated `@params.setter`?

Reply:
I like the idea of `params` being a property. It could be implemented in such a way that it transparently fills in default parameters before returning a dict. And it could have a setter used to set manual parameters (which would be stored in an internal field, say `_params`). Actually, the filling in could also be done within the setter - then it would only happen once instead of on every access. Accessing the `params` property would then just return `self._params` or perhaps a `copy()` to be safe.